onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -6725,10 +6725,46 @@ def aten_unflatten(self: TReal, dim: INT64, sizes: INT64):
     return op.Reshape(self, final_shape)
 
 
-def aten_unfold(self: TensorType, dimension: int, size: int, step: int) -> TensorType:
+@torch_op("aten::unfold", trace_only=True)
+def aten_unfold(self: TTensor, dimension: int, size: int, step: int) -> TTensor:
     """unfold(Tensor(a) self, int dimension, int size, int step) -> Tensor(a)"""
 
-    raise NotImplementedError()
+    self_rank = len(self.shape)
+    if self_rank == 0:
+        result = op.Unsqueeze(self, 0)
+    else:
+        dim_size = self.shape[dimension]
+        target_end = (dim_size - size) // step + 1
+        if target_end > 1:  # the rank of final reuslt will be self_rank + 1
+            self_rank = self_rank + 1
+        # perm need to be list[int], so have to be generated in trace_only mode
+        perm = list(range(self_rank))
+        perm.append(perm.pop(dimension + 1))
```

Comment:
Would be great to add a comment to explain this logic, thanks!

Reply:
add comment.

Reply:
Why do we do this though? What does it mean? Is there a reference logic?

Reply:
1.  similar but different in detail: https://github.com/pytorch/pytorch/blob/a8f40b39ce4f9fa9ffd90400b7d10ea4051d623a/torch/onnx/symbolic_opset12.py#L390
2. the same logic: https://github.com/microsoft/onnxscript/blob/50c3ee968960f9f7735df3f0f2458643d9d9ba08/onnxscript/function_libs/torch_lib/ops/core.py#L5999